internals/overlord/state/identities.go

```diff
@@ -47,27 +52,46 @@ type LocalIdentity struct {
 	UserID uint32
 }
 
+// BasicIdentity holds identity configuration specific to the "basic" type
+// (for HTTP basic authentication).
+type BasicIdentity struct {
+	Password string // Note: In a real application, store a password hash, not the plaintext password.
+}
+
 // validate checks that the identity is valid, returning an error if not.
 func (d *Identity) validate() error {
 	if d == nil {
 		return errors.New("identity must not be nil")
 	}
 
+	if d.Name != "" {
+		invalidChars := regexp.MustCompile(`[^a-zA-Z0-9_\-]`)
+		if invalidChars.MatchString(d.Name) {
+			return fmt.Errorf("identity name %q contains invalid characters (only alphanumeric, underscore, and hyphen allowed)", d.Name)
+		}
+	}
+
 	switch d.Access {
-	case AdminAccess, ReadAccess, UntrustedAccess:
+	case AdminAccess, ReadAccess, MetricsAccess, UntrustedAccess:
 	case "":
-		return fmt.Errorf("access value must be specified (%q, %q, or %q)",
-			AdminAccess, ReadAccess, UntrustedAccess)
+		return fmt.Errorf("access value must be specified (%q, %q, %q, or %q)",
+			AdminAccess, ReadAccess, MetricsAccess, UntrustedAccess)
 	default:
-		return fmt.Errorf("invalid access value %q, must be %q, %q, or %q",
-			d.Access, AdminAccess, ReadAccess, UntrustedAccess)
+		return fmt.Errorf("invalid access value %q, must be %q, %q, %q, or %q",
+			d.Access, AdminAccess, ReadAccess, MetricsAccess, UntrustedAccess)
 	}
 
 	switch {
 	case d.Local != nil:
+		return nil
+	case d.Basic != nil:
+		if d.Basic.Password == "" {
```

Comment:
Per our spec discussion, I believe that for now we only wanted to allow `access: metrics` (or I guess `untrusted`) with basic authentication, not read or admin.

Reply:
Is this comment intended for the `CheckAccess` method, the same as the previous comment?